src/commitlint/cli.py

```diff
@@ -119,36 +126,46 @@ def _get_commit_message_from_file(filepath: str) -> str:
         return commit_message
 
 
-def _handle_commit_message(commit_message: str, skip_detail: bool) -> None:
+def _handle_commit_message(
+    commit_message: str, skip_detail: bool, quiet: bool = False
+) -> None:
     """
     Handles a single commit message, checks its validity, and prints the result.
 
     Args:
         commit_message (str): The commit message to be handled.
         skip_detail (bool): Whether to skip the detailed error linting.
+        quiet (bool): Whether to ignore stout and stderr
 
     Raises:
         SystemExit: If the commit message is invalid.
     """
     success, errors = lint_commit_message(commit_message, skip_detail=skip_detail)
 
+    if quiet and not success:
+        sys.exit(1)
+
     if success:
-        sys.stdout.write(f"{VALIDATION_SUCCESSFUL}\n")
-    else:
-        _show_errors(commit_message, errors, skip_detail=skip_detail)
+        if not quiet:
+            sys.stdout.write(f"{VALIDATION_SUCCESSFUL}\n")
+
+    if not success:
+        if not quiet:
```

Comment:
suggestion: This check is redundant, we have already checked `quiet and not success`. Perhaps using just `else` will work.

Reply:
My bad. The test will fail because sys.exit is mocked and won't exit. Let's keep this code as it is. We can address handling sys.exit in tests later to improve our code.